language/move-prover/spec-lang/src/translate.rs

```diff
@@ -140,6 +142,14 @@ struct FunEntry {
     result_type: Type,
 }
 
+#[derive(Debug, Clone)]
+struct ConstEntry {
+    loc: Loc,
+    module_id: ModuleId,
+    const_id: ConstId,
+    signature: Type,
```

Comment:
Do not call types 'signature', we don't use that naming in this code. Call it `ty: Type`.

Also I'm wondering why we don't inline constants already here in translate.rs, and never add them to the GlobalEnv. This would be much easier:

1. Attach the translated ast::Value to ConstEntry
2. Populate the ConstEntry during decl_ana.
3. During def_ana, when expressions are translated, if you come along a name which resolves as a ConstEntry, just inline its value.

There is little merit to us to have the ConstDecls available in subsequent phases. This would simplify this implementation significantly.

Reply:
Yeah we could do what you suggested. I thought maybe it would be nice to still have the names of the constants at Boogie level. Since you think there's little merit to it, I'll change it.

Reply:
Note, just a suggestion.

Reply:
Right I think your suggestion makes sense. There's no need to carry constant name all the way down.